components/dashboard/dashboard.go

```diff
@@ -127,7 +130,7 @@ func (d *Dashboard) ReconcileComponent(cli client.Client, owner metav1.Object, d
 		}
 
 		// Update image parameters (ODH does not use this solution, only downstream)
-		if dscispec.DevFlags.ManifestsUri == "" && len(d.DevFlags.Manifests) == 0 {
+		if d.DevFlags != nil && dscispec.DevFlags != nil && dscispec.DevFlags.ManifestsUri == "" && len(d.DevFlags.Manifests) == 0 {
```

Comment:
this part i am not following:
1. DSCI devflags does not set ManifetsUri 
2. DSCI devFlags is set  
3. component devFlags is set
4. component devFlags does not set ManifestsUri
how could 1 and 2 happen at the same time  and 3 and 2 happen at the same time.
should that be just 
if d.DevFlags == nil && dscispec.DevFlags == nil  { // do it...}

Reply:
since we have added a `*DevFlags` to the DSCI as well, it will be nil unless the user sets it in the yaml
So in this case dscispec.DevFlags.ManifestUri will result in error because DevFlags will be nil( it will be like dscispec.nil.ManifestsUri) so that is why I added the check. Or should we not add a pointer to the DevFlags in DSCI? wdyt

Reply:
Hmm, does the original code mean "If neither dsci nor componet's manifests provided"? (not_dsci && not_component)

then one is not provided if the pointer is nil or the value is let's say blank.
not_dsci := (dscispec.DevFlags == nil || dscispec.DevFlags.ManifestsUri == "")
not_component := (d.DevFlags == nil || len(d.DevFlags.Manifests) == 0)

Reply:
let' break down it into 2 pieces, DSC and DSCI.
both are having the same logic in this if-condition.

what should trigger the caller of deploy.ApplyParams() is when user does not set DSC devFlags ManifestUri and user does not set component Devflags Manifests.

is this matching your implementation?

Reply:
something like this then
`		if (dscispec.DevFlags == nil || dscispec.DevFlags.ManifestsUri == "") && (c.DevFlags == nil || len(c.DevFlags.Manifests) == 0) {
			if err := deploy.ApplyParams(CodeflarePath+"/bases", c.SetImageParamsMap(imageParamMap), true); err != nil {
				return err
			}
		}`

Reply:
used this way now `(not_dsci && not_component)`